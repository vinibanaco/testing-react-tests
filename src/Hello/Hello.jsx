import React, { Component } from 'react';
import PropTypes from 'prop-types';

import InputBox from '../components/InputBox/InputBox';

class Hello extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      visible: true,
    };
  }

  componentDidMount() {
    const { name } = this.props;
    this.setState({ name });
  }

  toggleText = () => {
    const { visible } = this.state;
    this.setState({
      visible: !visible,
    });
  }

  changeTitle = (text) => {
    this.setState({
      name: text,
    });
  }

  renderTitle = () => {
    const { visible, name } = this.state;

    if (visible) {
      return <h1>{name}</h1>;
    }

    return null;
  }

  render() {
    return (
      <>
        {this.renderTitle()}
        <button type="button" onClick={this.toggleText}>Toggle text</button>

        <InputBox style={{ marginTop: '40px' }} callback={this.changeTitle} />
      </>
    );
  }
}

export default Hello;

Hello.propTypes = {
  name: PropTypes.string,
};

Hello.defaultProps = {
  name: 'Vin√≠cius',
};
