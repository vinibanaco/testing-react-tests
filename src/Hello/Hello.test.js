import React from 'react';
import { shallow } from 'enzyme';

import Hello from './Hello';

describe('Hello', () => {
  // Normal render with no props
  it('should render correctly with no props', () => {
    const component = shallow(<Hello />);

    expect(component).toMatchSnapshot();
  });

  // Render with the "name" prop
  it('should render correctly with "name" prop', () => {
    const component = shallow(<Hello name="Roberto" />);

    expect(component).toMatchSnapshot();
  });

  // Toggle the text
  it('should toggle the text on button click', () => {
    const component = shallow(<Hello />);

    component.find('button').simulate('click');
    expect(component).toMatchSnapshot();

    component.find('button').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
