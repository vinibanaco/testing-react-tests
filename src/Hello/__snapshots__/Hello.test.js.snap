// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello should render correctly with "name" prop 1`] = `
<Fragment>
  <h1>
    Roberto
  </h1>
  <button
    onClick={[Function]}
    type="button"
  >
    Toggle text
  </button>
  <InputBox
    callback={[Function]}
    style={
      Object {
        "marginTop": "40px",
      }
    }
  />
</Fragment>
`;

exports[`Hello should render correctly with no props 1`] = `
<Fragment>
  <h1>
    Vinícius
  </h1>
  <button
    onClick={[Function]}
    type="button"
  >
    Toggle text
  </button>
  <InputBox
    callback={[Function]}
    style={
      Object {
        "marginTop": "40px",
      }
    }
  />
</Fragment>
`;

exports[`Hello should toggle the text on button click 1`] = `
<Fragment>
  <button
    onClick={[Function]}
    type="button"
  >
    Toggle text
  </button>
  <InputBox
    callback={[Function]}
    style={
      Object {
        "marginTop": "40px",
      }
    }
  />
</Fragment>
`;

exports[`Hello should toggle the text on button click 2`] = `
<Fragment>
  <h1>
    Vinícius
  </h1>
  <button
    onClick={[Function]}
    type="button"
  >
    Toggle text
  </button>
  <InputBox
    callback={[Function]}
    style={
      Object {
        "marginTop": "40px",
      }
    }
  />
</Fragment>
`;
